### ISSUE TITLE
importAddress (with rescan) on doichain-cli throws callback is not a function in validateAddress

### ISSUE THREAD

**Issue by inspiraluna** [Doichain/dapp#149]
When manually importing a Doichain address via importAddress (with rescan) Doichain dApp throws an error inside validateAddress

 ```
got txid from memcache  a4748f013d3d095c9c31f2f2374234c0cbcd39d24402bf0cb91d868c49bd4102
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]: checking txs
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]: checking tx a4748f013d3d095c9c31f2f2374234c0cbcd39d24402bf0cb91d868c49bd4102
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]: validateaddress: {
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:   address: [Function],
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:   err: Error: JSON value is not a string as expected
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:       at /home/doichain/dapp/build/bundle/programs/server/npm/node_modules/meteor/doichain_doichain-meteor-api/node_modules/namecoin/lib/jsonrpc.js:104:19
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:       at Array.forEach (<anonymous>)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:       at IncomingMessage.<anonymous> (/home/doichain/dapp/build/bundle/programs/server/npm/node_modules/meteor/doichain_doichain-meteor-api/node_modules/namecoin/lib/jsonrpc.js:101:15)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:       at IncomingMessage.emit (events.js:322:22)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:       at IncomingMessage.EventEmitter.emit (domain.js:482:12)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:       at endReadableNT (_stream_readable.js:1187:12)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:       at processTicksAndRejections (internal/process/task_queues.js:84:21) {
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     code: -1
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:   },
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:   client: Client { rpc: Client { opts: [Object], http: [Object] } }
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]: }
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]: meteor://💻app/packages/doichain_doichain-meteor-api.js:8767
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     callback(err, data);
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     ^
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]: TypeError: callback is not a function
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at packages/doichain:doichain-meteor-api/server/api/doichain.js:51:9
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at /home/doichain/dapp/build/bundle/programs/server/npm/node_modules/meteor/doichain_doichain-meteor-api/node_modules/namecoin/lib/index.js:43:5
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at /home/doichain/dapp/build/bundle/programs/server/npm/node_modules/meteor/doichain_doichain-meteor-api/node_modules/namecoin/lib/jsonrpc.js:108:13
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at Array.forEach (<anonymous>)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at IncomingMessage.<anonymous> (/home/doichain/dapp/build/bundle/programs/server/npm/node_modules/meteor/doichain_doichain-meteor-api/node_modules/namecoin/lib/jsonrpc.js:101:15)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at IncomingMessage.emit (events.js:322:22)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at IncomingMessage.EventEmitter.emit (domain.js:482:12)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at endReadableNT (_stream_readable.js:1187:12)
May 13 20:15:17 doichain-v0-0-9-85 dapp-start.sh[5648]:     at processTicksAndRejections (internal/process/task_queues.js:84:21)
May 13 20:15:17 doichain-v0-0-9-85 systemd[1]: dapp.service: Main process exited, code=exited, status=1/FAILURE
```

**Comment by inspiraluna** [comment]
Turns out:

check_new_transactions.js:63 (when iterating over the recent txs) takes the details arrayList
but in our case it is empty: (!?!)
```
{
  "amount": 0.00000000,
  "confirmations": 22,
  "blockhash": "5f9b030826f4edb571216ff9b0403109e465a6c024464e6678445b54ffccb3e9",
  "blockindex": 1,
  "blocktime": 1589375416,
  "txid": "a4748f013d3d095c9c31f2f2374234c0cbcd39d24402bf0cb91d868c49bd4102",
  "walletconflicts": [
  ],
  "time": 1589375419,
  "timereceived": 1589375419,
  "bip125-replaceable": "no",
  "details": [
  ],
  "hex": "01000000015cb840056ae0b22a851ae9a0a8d8a9bcf1d8ea275b31a6150cf970825e0d914e000000008a47304402207829142707503f86aff03d35cf5b240b5d0c1aa62abdf84870fd43d5a647199a02204cdd93e2a24f931754da27bb6f3e6e8a877ef4816960f57a488dbbec49147c3f0141047c1f850b9d6d1700a08f373c85d4b0ef6834a37a315d95028cf7e2eea9fae99f7bda5afca36dfd88cbb7ea90f91dec864ca79fddbeb2f8420b73ced2bc7e783bfeffffff0210270000000000001976a91455b8add7ae5391eacddc0d55f08c26659110674b88ac00a60f00000000001976a91491a6d484b2fed402a5b72eb60dc4e03a4003f49a88ac92130200"
}
```

**Comment by inspiraluna** [comment]
Turns out further: the input is a coinbase transaction so 
addCoinTx:208 runs into a problem while extracing the publicKey / address from the input.
Now: creating a hotfix release while checking if a publicKey is available before calling validateAddress. (with no 'real' address)              

`if (inTx.scriptSig &&  inTx.scriptSig.asm.indexOf('[ALL] ') !==-1 ) { //try to get the public key from the input of this transaction `

**Comment by inspiraluna** [comment]
Creating release v0.0.9.86 and deploying it.

**Comment by inspiraluna** [comment]
solved in v0.0.9.86 see: 
- dApp: https://doichain-v0-0-9-86.doi.works/
- doiContact Wallet: https://ipfs.io/ipfs/QmecAsS21P3tch1EAZ5cpxyMYncjaZjkWdnfAe74spLEbs/
